GVFS/GVFS.Common/GVFSEnlistment.cs

```diff
@@ -11,9 +11,9 @@ namespace GVFS.Common
     public partial class GVFSEnlistment : Enlistment
     {
         public const string BlobSizesCacheName = "blobSizes";
+        public const string InvalidRepoUrl = "invalid://repoUrl";

```

Comment:
I don't think we should make this constant public. The point of it was that it should never leak out of this class.

Right now, I think you're fixing the symptom by checking for this constant before displaying it. But how did the invalid URL leak out of here in the first place? What other code paths might be affected? Rather than special case all uses, I'd like to understand the root cause better.